newspaper/views.py

```diff
@@ -69,6 +69,59 @@ def register():
     return render_template("signup.html", error=error)
 
 
+@app.route("/leave", methods=("POST", "GET"))
+def leave():
+    error = None
+    if request.method == "POST":
+        # Get and validate the email
+        email = request.form["email"]
+        try:
+            validate_email(email)
+        except EmailNotValidError:
+            error = "Invalid email"
+
+        # Check if the email is already used
+        if not User.query.filter_by(email=email).first():
+            return "<h1>There is no user with this email.</h1>"
+
+        # Create the token and the confirmation link
+        token = serializer.dumps(email)
+
+        # Create and send the confirmation message
+        msg = Message(
+            "Confirm Email",
+            recipients=[email],
+            body=f"Token: {token}",
+        )
+        mail.send(msg)
+
+        return redirect(url_for("confirm"))
+
+    return render_template("leave.html", error=error)
+
+
+@app.route("/confirm", methods=("POST", "GET"))
```

Comment:
There is already a `/confirm-email`, that is a little bit confusing..

Reply:
yeah that needs cleaning up but they are quite different